spatialpandas/tools/sjoin.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+import pandas as pd
+
+
+def _record_reset_index(df, suffix):
+    # Record original index name(s), generate new index name(s), reset index column(s)
+    new_column_name = ["index_%s" % suffix]
+    df = df.copy(deep=True)
+    try:
+        old_index_name = [df.index.name]
+        df.index = df.index.rename(new_column_name[0])
+    except TypeError:
+        new_column_name = [
+            "index_%s" % suffix + str(l) for l, ix in
+            enumerate(df.index.names)
+        ]
+        old_index_name = df.index.names
+        df.index = df.index.rename(new_column_name)
+    df = df.reset_index(drop=False)
+
+    return df, old_index_name, new_column_name
+
+
+def sjoin(
+        left_df, right_df, how="inner", op="intersects",
+        lsuffix="left", rsuffix="right"
+):
+    """
+    Spatial join between two GeoDataFrames or between a DaskGeoDataFrame and
+    a GeoDataFrame
+
+    Args:
+        left_df: A GeoDataFrame or DaskGeoDataFrame
+        right_df: A GeoDataFrame
+        how: The type of join. One of 'inner', 'left', or 'right'. Note that 'right'
+            is not supported when left_df is a DaskGeoDataFrame
+        op: Binary predicate, currently only "intersects" is supported
+        lsuffix: Suffix to apply to overlapping column names from the left GeoDataFrame
+        rsuffix: Suffix to apply to overlapping column names from the right GeoDataFrame
+
+    Returns:
+        GeoDataFrame or DaskGeoDataFrame
```

Comment:
Where the type returned depends on left_df?

Reply:
:+1: